lib/ecl/well_conn.cpp

```diff
@@ -157,73 +154,85 @@ well_conn_type * well_conn_alloc_from_kw( const ecl_kw_type * icon_kw ,
                                           int conn_nr ) {
 
   const int icon_offset = header->niconz * ( header->ncwmax * well_nr + conn_nr );
-  int IC = ecl_kw_iget_int( icon_kw , icon_offset + ICON_IC_INDEX );
-  if (IC > 0) {
-    well_conn_type * conn;
-    int i       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_I_INDEX ) - 1;
-    int j       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_J_INDEX ) - 1;
-    int k       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_K_INDEX ) - 1;
-    double connection_factor = -1;
-    bool matrix_connection = true;
-    bool open;
-    well_conn_dir_enum dir = well_conn_fracX;
-
-    /* Set the status */
-    {
-      int int_status = ecl_kw_iget_int( icon_kw , icon_offset + ICON_STATUS_INDEX );
-      if (int_status > 0)
-        open = true;
-      else
-        open = false;
-    }
+  int IC =ecl_kw_iget_int( icon_kw , icon_offset + ICON_IC_INDEX );
+  if (IC <= 0)
+    return NULL;  /* IC < 0: Connection not in current LGR. */
 
 
-    /* Set the K value and fracture flag. */
-    {
-      if (header->dualp) {
-        int geometric_nz = header->nz / 2;
-        if (k >= geometric_nz) {
-          k -= geometric_nz;
-          matrix_connection = false;
-        }
-      }
-    }
+  /*
+    Out in the wild we have encountered files where some of integer values used
+    to indicate direction have had invalid values. In this case we just return
+    NULL and forget about the connection - it seems to work.
+  */
+  int int_direction = ecl_kw_iget_int( icon_kw , icon_offset + ICON_DIRECTION_INDEX );
+  if ((int_direction < 0) || (int_direction > ICON_FRACY)) {
+    fprintf(stderr,"Invalid direction value:%d encountered for well - connection ignored\n",int_direction);
+    return NULL;
+  }
+
+
+  int i       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_I_INDEX ) - 1;
+  int j       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_J_INDEX ) - 1;
+  int k       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_K_INDEX ) - 1;
+  double connection_factor = -1;
+  bool matrix_connection = true;
+  bool open;
+  well_conn_dir_enum dir = well_conn_fracX;
+
+  /* Set the status */
+  {
+    int int_status = ecl_kw_iget_int( icon_kw , icon_offset + ICON_STATUS_INDEX );
+    if (int_status > 0)
+      open = true;
+    else
+      open = false;
+  }
```

Comment:
These lines could perhaps be replaced with
```c
bool open = ecl_kw_iget_int(icon_kw, icon_offset + ICON_STATUS_INDEX) > 0
```

Reply:
Perhaps we should rename `open` to something else

Reply:
Have replaced the `open` assignment with your suggestion. The name `open` is quite descriptive in my opinion - so I have left that.

Reply:
The point wasn't that it wasn't descriptive, but I thought maybe it wasn't considered in good taste to reuse standard library names.

Reply:
OK - I agree; have changed to `is_open`